app.js

```diff
@@ -1,121 +1,122 @@
-  savedData = JSON.parse(savedData);
-  const dataUserObject = {
-    name: author.value,
-    title: title.value,
-  };
-  if (savedData !== null) {
-    savedData.push(dataUserObject);
-    localStorage.setItem('BookData', JSON.stringify(savedData));
-  } else {
-    NbookData.push(dataUserObject);
-    localStorage.setItem('BookData', JSON.stringify(NbookData));
+class Store {
+  static getBooks() {
+    let books;
+    if (localStorage.getItem('books') === null) {
+      books = [];
+    } else {
+      books = JSON.parse(localStorage.getItem('books'));
+    }
+    return books;
   }
-};
 
-/* button form */
-const addNewBook = document.querySelector('.add-btn');
-addNewBook.addEventListener('click', () => {
-  newBook();
-  addUserData();
-});
+  static addBook(book) {
+    const books = Store.getBooks();
+    books.push(book);
+    localStorage.setItem('books', JSON.stringify(books));
+  }
 
-/* display data */
-
-function getUserData() {
-  let data = localStorage.getItem('BookData');
-  data = JSON.parse(data);
-  if (data !== null) {
-    for (let j = 0; j < data.length; j += 1) {
-      const deck = document.getElementById('books-container');
-      const bookData = document.createElement('div');
-      const bookTitle = document.createElement('p');
-      const bookAuthor = document.createElement('p');
-      const removeBtn = document.createElement('button');
-      const line = document.createElement('div');
-      const booksaved = data[j].title;
-      bookData.className = 'book';
-      deck.appendChild(bookData);
-
-      bookTitle.classList.add('book-title');
-      bookAuthor.classList.add('book-author');
-      removeBtn.classList.add('remove-btn');
-      line.classList.add('line');
-
-      bookTitle.setAttribute('id', `bt${i}`);
-      bookAuthor.setAttribute('id', `ba${i}`);
-      removeBtn.setAttribute('id', `rb${i}`);
-      i += 1;
-      const btId = bookTitle.getAttribute('id');
-      const baId = bookAuthor.getAttribute('id');
-      const rbId = removeBtn.getAttribute('id');
-
-      bookData.appendChild(bookTitle);
-      bookData.appendChild(bookAuthor);
-      bookData.appendChild(removeBtn);
-      bookData.appendChild(line);
-
-      document.getElementById(btId).textContent = data[j].title;
-      document.getElementById(baId).textContent = data[j].name;
-      document.getElementById(rbId).textContent = 'Remove';
-
-      removeBtn.addEventListener('click', () => {
-        bookData.remove();
-        for (let k = 0; k < data.length; k += 1) {
-          if (booksaved === data[k].title) {
-            data.splice(k, 1);
-            localStorage.setItem('BookData', JSON.stringify(data));
-          }
-        }
-      });
-    }
+  static removeBook(author) {
+    const books = Store.getBooks();
+    books.forEach((book, index) => {
+      if (book.author === author) {
+        books.splice(index, 1);
+      }
+    });
+    localStorage.setItem('books', JSON.stringify(books));
```

Comment:
- [ ] The remove functionality doesn't seem to work properly as it only removes it from the display and not the localStorage. Please kindly make sure it actually removes the book from the localStorage, so that when the page is refreshed it gets the updated data.

Reply:
the problem is fixed. Could you check again please?